src/main/java/no/nav/vault/jdbc/hikaricp/HikariCPVaultUtil.java

```diff
@@ -41,6 +41,13 @@ public static HikariDataSource createHikariDataSourceWithVaultIntegration(final
             @Override
             public void run() {
                 try {
+                    if (hikariCPVaultUtil.ds != null && hikariCPVaultUtil.ds.isClosed()) {
+                        logger.info("Shutting down Timer");
+                        instance.getTimer().cancel();
```

Comment:
Trenger vel egentlig ikke denne ? 
Vi scheduler bare ikke timeren til å kjøre på nytt ved return

Reply:
Det stemmer - vi trenger kanskje ikke den cancel-en. Får krysse fingrene for at det ikke gjør noen skade!

Reply:
Den timeren har har en rask som refresher tokenet også vel?